filecoin-proofs/src/api/mod.rs

```diff
@@ -318,6 +323,153 @@ where
     add_piece(source, target, piece_size, Default::default())
 }
 
+// Verifies if a DiskStore specified by a config is consistent.
+fn verify_store(config: &StoreConfig, arity: usize) -> bool {
+    let store_path = StoreConfig::data_path(&config.path, &config.id);
+    if !Path::new(&store_path).exists() {
+        return false;
+    }
+
+    if config.size.is_none() {
+        return false;
+    }
+
+    let res = DiskStore::<<DefaultPieceHasher as Hasher>::Domain>::is_consistent(
+        config.size.unwrap(),
+        arity,
+        &config,
+    )
+    .is_ok();
+    trace!("disk store {:?} is_consistent: {}", store_path, res);
+
+    res
+}
+
+// Checks for the existence of the tree d store, the replica, and all generated labels.
+pub fn validate_cache_for_precommit_phase2<R, T>(
+    cache_path: R,
+    replica_path: T,
+    seal_precommit_phase1_output: &SealPreCommitPhase1Output,
+) -> Result<bool>
+where
+    R: AsRef<Path>,
+    T: AsRef<Path>,
+{
+    let replica = replica_path.as_ref().to_path_buf();
+    if !Path::new(&replica).exists() {
+        trace!("replica {:?} does not exist", &replica);
+        return Ok(false);
+    }
+
+    for label in &seal_precommit_phase1_output.labels.labels {
+        let cache = cache_path.as_ref().to_path_buf();
+        let current_path = StoreConfig::data_path(&label.path, &label.id);
+        if !verify_store(label, BINARY_ARITY) || !current_path.starts_with(&cache) {
+            trace!(
+                "store path [{:?}] and cache_path [{:?}] do not match",
+                &current_path,
+                &cache
+            );
+            return Ok(false);
+        }
+    }
+
+    Ok(verify_store(
+        &seal_precommit_phase1_output.config,
+        BINARY_ARITY,
```

Comment:
is this only used for comm_d?

Reply:
Yes.  At that stage, I believe only labels and comm_d are present.